smallrye-reactive-messaging-pulsar/src/main/java/io/smallrye/reactive/messaging/pulsar/tracing/PulsarAttributesExtractor.java

```diff
@@ -57,12 +57,12 @@ public String getConversationId(PulsarTrace pulsarTrace) {
 
         @Override
         public Long getMessagePayloadSize(PulsarTrace pulsarTrace) {
-            return null;
+            return pulsarTrace.getUncompressedPayloadSize();
         }
 
         @Override
         public Long getMessagePayloadCompressedSize(PulsarTrace pulsarTrace) {
-            return pulsarTrace.getUncompressedPayloadSize();
+            return null;
```

Comment:
can't we have the data here?

Reply:
I'll have a look, rather not through a direct API call.

Reply:
The compression is set on the client/producer, the message size is taken from the Pulsar Message object. There is no access to the information (not an easy one).